backend/api/src/websockets/switchboard.ts

```diff
@@ -0,0 +1,65 @@
+import { WebSocket } from 'ws'
+
+export type ClientState = {
+  uid?: string
+  lastSeen: number
+  subscriptions: Set<string>
+}
+
+/** Tracks the relationship of clients to websockets and subscription lists. */
+export class Switchboard {
```

Comment:
Classes? Hmm... I guess it's fine. Manifold employees might not be familiar with all the nuances of 'this' in classes.

Reply:
I guess on reflection "Manifold house style" is to just make some global variables that aren't in classes?

In Javascript if I want a bag of mutable state that I need to operate on repeatedly then I typically make a class and instantiate the class.

It's funny that you mention `this`, since probably the main reason Javascript introduced classes at all, was because it was harder to understand the implementation of `this` in the old "explicitly add functions to a prototype" or "use a closure like a class" styles that people would use in circa-2010 JS.

Reply:
One style is `initX()` which gives you the state, and then a bunch of functions that take x as their first parameter and mutate it.

Another style which is cool would be to have a function `createX()` which returns an object with all the methods. This has the downside of recreating the methods for every instantiation, whereas the class creates them once on the prototype object. (But I don't think that matters much here.)

E.g. with the latter style
```
function createSwitchboard() {
  const state = { ... }
  
  return {
    method1: () => {},
    method2: () => {}
  }
}
```